ownCloud/Settings/SecuritySettingsSection.swift

```diff
@@ -0,0 +1,161 @@
+//
+//  SecurityGlobalSettings.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 30/04/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import LocalAuthentication
+
+// MARK: - Security UserDefaults keys
+public let SecuritySettingsfrequencyKey: String =  "security-settings-frequency"
+public let SecuritySettingsPasscodeKey: String = "security-settings-usePasscode"
+public let SecuritySettingsBiometricalKey: String = "security-settings-useBiometrical"
+
+// MARK: - Section identifier
+private let SecuritySectionIdentifier: String = "settings-security-section"
+
+// MARK: - Row identifiers
+private let SecurityFrequencyRowIdentifier: String = "security-frequency-row"
+private let SecurityPasscodeRowIdentifier: String = "security-passcode-row"
+private let SecurityBiometricsRowIdentifier: String = "security-biometrical-row"
+
+// MARK: - SecurityAskfrequency
+enum SecurityAskfrequency: String {
+    case allways = "Allways"
+    case oneMinute = "1 minute"
+    case fiveMinutes = "5 minutes"
+    case thirtyMinutes = "30 minutes"
+
+    static let all = [allways, oneMinute, fiveMinutes, thirtyMinutes]
+}
+
+class SecuritySettings: StaticTableViewSection {
+
+    // MARK: - Security settings properties
+
+    /// Time between avery ask for security
+    private var frequency: SecurityAskfrequency
+    /// Passcode protection enabled
+    private var passcodeEnabled: Bool
+    /// Biometrical protection enabled
+    private var biometricalSecurityEnabled: Bool
+    /// UserDefaults used to store and retrieve all.
+    private var userDefaults: UserDefaults
+
+    init(userDefaults: UserDefaults) {
+        self.userDefaults = userDefaults
+
+        if let frequencySetting = userDefaults.string(forKey: SecuritySettingsfrequencyKey), let frequencyValue = SecurityAskfrequency(rawValue: frequencySetting) {
+            self.frequency = frequencyValue
+        } else {
+            self.frequency = .allways
+        }
+        self.passcodeEnabled = userDefaults.bool(forKey: SecuritySettingsPasscodeKey)
+        self.biometricalSecurityEnabled = userDefaults.bool(forKey: SecuritySettingsBiometricalKey)
+
+        super.init()
+        self.headerTitle = "Security".localized
+        self.identifier = SecuritySectionIdentifier
+
+        updateUI()
```

Comment:
Instead of breaking the creation of the rows into methods, and then checking if they exist using their identifier lateron, you could generate all rows here in the init-method and save them in instance variables.

Benefits:
- you don't need to keep track of whether or not they were already created (=> they're all just there already, ready for use).
- you can use the row objects directly and don't have to find and fetch them through their identifiers (=> you also don't need to generate symbols for the identifiers (like f.ex. `SecurityBiometricsRowIdentifier`) then, simplifying the implementation further)
- to change the interface dynamically, you could either keep using add/remove methods on the `StaticTableViewSection` - or put together a new `rows` array and ask the tableView to reload just that section. Ideally, you could extend `StaticTableViewSection` with a "`replaceRows`" method that replaces the rows instance var with a passed in array, and triggers a section reload. I also noticed that `StaticTableViewSection` doesn't support "live" additions/removals and animation yet, so it may be a good idea to:
   - add animation support to `add(rows:)` (like `add(rows:animated:)` where animated defaults to false and only calls `insertRows(at indexPaths: [IndexPath], with animation: UITableViewRowAnimation)` if the `viewController` property is set)
   - add a `remove(rows:animated:)` method that does the same, just for removal
   - add `add(row:animated:)` and `remove(row:animated:)` convenience methods that wrap the passed row into an array and forward it to the `add(rows:animated:)` and `remove(rows:animated:)` methods.
   - in the future, an `insert(rows:indexes:animated:)` method could be added, too - and `add(rows:animated:)` be changed to become just a convenience method to call that. But if it's not needed at the moment, this can surely be saved until we need it.